File: Random experiments/get_filtered_signal.m
function ffilt =  get_filtered_signal(filename)
Fs = 0;
y = zeros(1,10000000);
[y,Sampling_frequency] = audioread(filename);

Fs = Sampling_frequency;
Filter_limit = 1;


PSD = zeros(1,1000);
x = zeros(1,1486272);


Fn = Fs/2;
dt = 1/Fs;

 i = 1; 
 j = 1;

    for i = 1:length(y)
        if y(i) ~= 0
        
         x(j) = y(i);
         j = j + 1;
        
        end     
    end
    
    n = length(x);
    fhat = fft(x,n);
    
    PSD = fhat.*conj(fhat)/n;
    indices = PSD>Filter_limit;
    PSDclean = PSD.*indices;
    fhat = indices.*fhat;
    
    ffilt = ifft(fhat);
    
    

    
end

File: Random experiments/codegen/lib/get_filtered_signal/get_filtered_signal.c
/*
 * Academic License - for use in teaching, academic research, and meeting
 * course requirements at degree granting institutions only.  Not for
 * government, commercial, or other organizational use.
 *
 * get_filtered_signal.c
 *
 * Code generation for function 'get_filtered_signal'
 *
 */

/* Include files */
#include "get_filtered_signal.h"
#include "fft.h"
#include "ifft.h"
#include <string.h>

/* Function Definitions */
void get_filtered_signal(const double y[23085504], double Sampling_frequency,
  creal_T ffilt[1486272])
{
  static double x[1486272];
  int j;
  int i;
  static creal_T fhat[1486272];
  double fhat_re;
  (void)Sampling_frequency;
  memset(&x[0], 0, 1486272U * sizeof(double));
  j = 0;
  for (i = 0; i < 11542752; i++) {
    if (y[i] != 0.0) {
      x[j] = y[i];
      j++;
    }
  }

  fft(x, fhat);
  for (j = 0; j < 1486272; j++) {
    fhat_re = fhat[j].re * fhat[j].re - fhat[j].im * -fhat[j].im;
    if (fhat[j].re * -fhat[j].im + fhat[j].im * fhat[j].re == 0.0) {
      fhat_re /= 1.486272E+6;
    } else if (fhat_re == 0.0) {
      fhat_re = 0.0;
    } else {
      fhat_re /= 1.486272E+6;
    }

    fhat[j].re *= (double)(fhat_re > 1.0);
    fhat[j].im *= (double)(fhat_re > 1.0);
  }

  ifft(fhat, ffilt);
}

/* End of code generation (get_filtered_signal.c) */
